File: running-app-backend/account/models/notification_setting.py
import uuid
from django.db import models

class NotificationSetting(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, db_index=True)
    user = models.OneToOneField('account.User', related_name="notification_setting", on_delete=models.CASCADE)
    finished_workout = models.BooleanField(default=True)
    comment = models.BooleanField(default=True)
    like = models.BooleanField(default=True)
    mentions_on_activities = models.BooleanField(default=True)
    respond_to_comments = models.BooleanField(default=True)
    new_follower = models.BooleanField(default=True)
    following_activity = models.BooleanField(default=True)
    request_to_follow = models.BooleanField(default=True)
    approved_follow_request = models.BooleanField(default=True)
    pending_join_requests = models.BooleanField(default=True)
    invited_to_club = models.BooleanField(default=True)

File: running_app/lib/models/account/notification_setting.dart
class NotificationSetting {
  final String? id;
  final bool? finishedWorkout;
  final bool? comment;
  final bool? like;
  final bool? mentionsOnActivities;
  final bool? respondToComments;
  final bool? newFollower;
  final bool? followingActivity;
  final bool? requestToFollow;
  final bool? approvedFollowRequest;
  final bool? pendingJoinRequests;
  final bool? invitedToClub;

  NotificationSetting({
    this.id,
    required this.finishedWorkout,
    required this.comment,
    required this.like,
    required this.mentionsOnActivities,
    required this.respondToComments,
    required this.newFollower,
    required this.followingActivity,
    required this.requestToFollow,
    required this.approvedFollowRequest,
    required this.pendingJoinRequests,
    required this.invitedToClub,
  });

  NotificationSetting.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        finishedWorkout = json['finished_workout'],
        comment = json['comment'],
        like = json['like'],
        mentionsOnActivities = json['mentions_on_activities'],
        respondToComments = json['respond_to_comments'],
        newFollower = json['new_follower'],
        followingActivity = json['following_activity'],
        requestToFollow = json['request_to_follow'],
        approvedFollowRequest = json['approved_follow_request'],
        pendingJoinRequests = json['pending_join_requests'],
        invitedToClub = json['invited_to_club'];


  Map<String, dynamic> toJson() {
    return {
      'finished_workout': finishedWorkout,
      'comment': comment,
      'like': like,
      'mentions_on_activities': mentionsOnActivities,
      'respond_to_comments': respondToComments,
      'new_follower': newFollower,
      'following_activity': followingActivity,
      'request_to_follow': requestToFollow,
      'approved_follow_request': approvedFollowRequest,
      'pending_join_requests': pendingJoinRequests,
      'invited_to_club': invitedToClub,
    };
  }

  @override
  String toString() {
    return '${toJson()}';
  }
}